### PR TITLE
Added radar customization with persistent saving

### PR DESCRIPTION
**Pull request by Ze-Rax**

### CODE REVIEW

File: DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt [hunk 1]
```diff
@@ -85,159 +50,471 @@ Text GetMarkersManialink() {
...
-				if (OffsetPercentage > {{{ C_MarkerFadeStart }}}) {
-					declare Real FadeProgress = ({{{ C_MarkerFadeEnd }}} - OffsetPercentage) / ({{{ C_MarkerFadeEnd - C_MarkerFadeStart }}});
-					Opacity = ML::Clamp(FadeProgress, 0., 1.);
+			}
+
+			/**
+			 * Persistently saves the radar config the player.
+			 */
+			Void SaveRadarUserConfig() {
+				declare persistent K_RadarUserConfig RadarConfig for LocalUser;
```
Comment (reaby) [inline DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt:245]: i think this shoudln't be persistent

File: DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt [hunk 2]
```diff
@@ -85,159 +50,471 @@ Text GetMarkersManialink() {
...
+			 */
+			 Vec2 ClampToScreen(Vec2 Position) {
+				return < ML::Clamp(Position.X, -160., 160.), ML::Clamp(Position.Y, -90., 90.) >;
+			}
+
+			/**
+			 * Updates the persistent settings
+			 */
+			 Void UpdateRadarSettings() {
+				declare persistent K_RadarUserConfig RadarConfig for LocalUser;
```
Comment (reaby) [inline DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt:263]: same here

File: DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt (hunk 1 above)
Comment (Ze-Rax) [reply DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt:245]: > i think this shoudln't be persistent

This is not saved. Only variables for MainUser are saved.

Comment (Ze-Rax) [reply DedicatedServer/Scripts/Libs/Zrx/ModeLibs/FlagRush/UI/Modules/Radar.Script.txt:245]: But ye, should not be marked as persistent here. Apparently script cache isn't a thing anymore. I gad an issue in my thinking. I Just wanted it to be preserved over reinits, so it's not reset if reinit happens during editing dialog. But a normal declare for should do the trick

**Review by reaby — APPROVED** [review]
I approve now.

**Pull request merged**